src/stcal/resample/resample.py

```diff
@@ -0,0 +1,1506 @@
+                        "filename",
+                        "level",
+                        "subtracted",
+                        "exposure_time",
+                    ]
+                )
+
+                # Check that input models are 2D images
+                if in_data.ndim != 2:
+                    raise RuntimeError(
+                        f"Input {attrs['filename']} is not a 2D image."
+                    )
+
+                input_pixflux_area = attrs["pixelarea_steradians"]
+                imwcs = attrs["wcs"]
+                if (input_pixflux_area and
+                        'SPECTRAL' not in imwcs.output_frame.axes_type):
+                    imwcs.array_shape = in_data.shape
+                    input_pixel_area = _compute_image_pixel_area(imwcs)
+                    if input_pixel_area is None:
+                        raise ValueError(
+                            "Unable to compute input pixel area from WCS of input "
+                            f"image {repr(attrs['filename'])}."
+                        )
+                    iscale = np.sqrt(input_pixflux_area / input_pixel_area)
+                else:
+                    iscale = 1.0
+
+                # TODO: should weight_type=None here?
+                in_wht = self.build_driz_weight(
+                    model,
+                    weight_type=self.weight_type,
+                    good_bits=self.good_bits
+                )
+
+                # apply sky subtraction
+                blevel = attrs["level"]
+                if not attrs["subtracted"] and blevel is not None:
+                    in_data = in_data - blevel
+
+                xmin, xmax, ymin, ymax = _resample_range(
+                    in_data.shape,
+                    imwcs.bounding_box
+                )
+
+                pixmap = calc_pixmap(wcs_from=imwcs, wcs_to=self._output_wcs)
+
+                add_image_kwargs = {
+                    'exptime': attrs["exposure_time"],
+                    'pixmap': pixmap,
+                    'scale': iscale,
+                    'weight_map': in_wht,
+                    'wht_scale': 1.0,
+                    'pixfrac': self.pixfrac,
+                    'in_units': 'cps',  # TODO: get units from data model
+                    'xmin': xmin,
+                    'xmax': xmax,
+                    'ymin': ymin,
+                    'ymax': ymax,
+                }
+
+                driz_data.add_image(in_data, **add_image_kwargs)
+
+                if self._enable_err:
+                    self._resample_variance_data(model, None, add_image_kwargs)
+
+                self._input_models.shelve(model, modify=False)
+
+        # TODO: see what to do about original update_exposure_times()
+
+        return self.create_output_model(driz_data)
+
+
+class ResampleSingle(ResampleBase):
+    """
+    This is the controlling routine for the resampling process.
+
+    Notes
+    -----
+    This routine performs the following operations::
```

Comment:
this looks identical to ResampleCoAdd.  Some doc updates should be made to clarify how they are different